### PR TITLE
Project 15 - Petra Almgren

### PR DESCRIPTION
**Pull request by petrame**
Link to my deployed project:
https://project-todos.netlify.app/

### CODE REVIEW

File: code/src/components/AddTodo.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        className="add-todo-text"
+        type="text"
+        placeholder="Write your task here"
+        value={value}
+        onChange={(event) => setValue(event.target.value)}
+        onKeyPress={handleKeypress}
+      />
+      <button
+        className="add-todo-button"
+        disabled={value.length < 2 || value.length > 150}
```
Comment (mats-johansen) [inline code/src/components/AddTodo.js:39]: This is nice!

File: code/src/components/ClearButtons.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    <div className="button-container">
+      <button className="clear-complete-button" onClick={onCompleteItemsDelete}>
+        Clear completed
+      </button>
+      <button className="clear-all-button" onClick={onAllItemsDelete}>
+        Clear all
+      </button>
+    </div>
+  );
+};
```
Comment (mats-johansen) [inline code/src/components/ClearButtons.js:28]: Great job collecting all remove/delete functions in one component

File: code/src/components/StatusBar.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    <section className="status-bar">
+      <p className="nr-of-tasks">
+        {items.length === 0
+          ? "Your todo list is empty"
+          : `${completedItems.length}/${items.length} tasks done`}
+      </p>
+      <ClearButtons />
+    </section>
+  );
+};
```
Comment (mats-johansen) [inline code/src/components/StatusBar.js:21]: The statusbar is a very nice component on your page, very user friendly!

File: code/src/css/addTodo.css [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+.add-todo-button:enabled:active {
+  transform: translateY(1px);
+  box-shadow: none;
+}
+
+.plus-sign {
+  font-size: 16px;
+}
```
Comment (mats-johansen) [inline code/src/css/addTodo.css:39]: Would have been nice with styled-components for practice purposes

File: code/src/reducer/todos.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    removeCompleted: (store, action) => {
+      const filteredList = store.items.filter((item) => !item.isComplete);
+      store.items = filteredList;
+    },
+
+    addItem: (store, action) => {
+      const setId = () => {
+        if (store.items.length === 0) return 1;
+        else return Math.max(...store.items.map((item) => item.id)) + 1;
```
Comment (mats-johansen) [inline code/src/reducer/todos.js:29]: Nice!

**Review by mats-johansen — COMMENTED** [review]
Great job on the todo list and I love the styling of the page! You finished the blue level and some more and an extra + for the checkboxes. They turned out great! I think you did very well on being consistent with the usage of ; and indents and it is clear that you cleaned up and looked over your code before handing it in.

File: code/src/css/todoItem.css [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+.checkmark {
+  width: 100%;
+  height: 100%;
+  border: 1px solid #063053;
+  border-radius: 3px;
+  display: inline-block;
+  background: #5c7c99 url("../img/checked.png") center/0 no-repeat;
+  transition: background-size 300ms ease-in;
+}
```
Comment (HIPPIEKICK) [inline code/src/css/todoItem.css:43]: If you write a long todo, the checkbox shrinks👇
Skärmavbild 2020-12-21 kl  13 45 29

**Pull request closed without merging**